packages/dart/npt_flutter/lib/features/profile_list/bloc/profile_list_bloc.dart

```diff
@@ -97,4 +101,128 @@ class ProfileListBloc extends LoggingBloc<ProfileListEvent, ProfileListState> {
 
     emit(ProfileListLoaded(profiles: profiles));
   }
+
+  /// Handles sorting of the profile list.
+  Future<void> _onSort(
+    ProfileListSortEvent event,
+    Emitter<ProfileListState> emit,
+  ) async {
+    App.log('Sort event received: column=${event.sortColumn}'.loggable);
+    // Don't allow sorts unless listed is loaded - this reduces the number of edge cases significantly
+    if (state is! ProfileListLoaded) {
+      App.log('Cannot sort - state is not ProfileListLoaded'.loggable);
+      return;
+    }
+    var currentState = state as ProfileListLoaded;
+    App.log(
+      'Current sort: column=${currentState.sortColumn}, order=${currentState.sortOrder}'
+          .loggable,
+    );
+    // Toggle sort order if clicking same column
+    final newSortOrder =
+        currentState.sortColumn == event.sortColumn &&
+            currentState.sortOrder == SortOrder.ascending
+        ? SortOrder.descending
+        : SortOrder.ascending;
```

Comment:
This ensure the first click on a new column always sorts the column in ascending order,